src/dbus_api/api.rs

```diff
@@ -64,10 +66,10 @@ fn get_next_arg<'a, T>(iter: &mut Iter<'a>, loc: u16) -> Result<T, MethodErr>
 
 /// Get filesystem name from object path
 fn object_path_to_pair(dbus_context: &DbusContext,
-                       fs_object_path: &str)
-                       -> Result<(String, String), (MessageItem, MessageItem)> {
-    let fs_pool_pair = match dbus_context.filesystems.borrow().get_by_first(fs_object_path) {
-        Some(fs_name) => fs_name.clone(),
+                       fs_object_path: &dbus::Path)
+                       -> Result<(Uuid, Uuid), (MessageItem, MessageItem)> {
```

Comment:
The method will go away. This was an intermediate commit, to get to a compiling state. The final version returned the object path of the pool and the uuid of the filesystem.